tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -643,6 +643,76 @@ def _check_rules(self, resource_rules):
             raise FilterValidationError("Internal error.")
 
 
+@six.add_metaclass(ABCMeta)
+class FirewallBypassFilter(Filter):
+    """Filters resources by the firewall bypass rules
+
+    """
+
+    @staticmethod
+    def schema(values):
+        return type_schema(
+            'firewall-bypass',
+            required=['mode', 'list'],
+            **{
+                'mode': {'enum': ['include', 'equal', 'any', 'only']},
+                'list': {'type': 'array', 'items': {'enum': values}}
+            })
+
+    log = logging.getLogger('custodian.azure.filters.FirewallRulesFilter')
+
+    def __init__(self, data, manager=None):
+        super(FirewallBypassFilter, self).__init__(data, manager)
+        self.mode = self.data['mode']
+        self.list = set(self.data['list'])
+        self.client = None
+
+    def process(self, resources, event=None):
+        self.client = self.manager.get_client()
+
+        result, _ = ThreadHelper.execute_in_parallel(
+            resources=resources,
+            event=event,
+            execution_method=self._check_resources,
+            executor_factory=self.executor_factory,
+            log=self.log
+        )
+
+        return result
+
+    def _check_resources(self, resources, event):
+        return [r for r in resources if self._check_resource(r)]
+
+    @abstractmethod
+    def _query_bypass(self, resource):
```

Comment:
Why is ` _query_bypass` using the private function naming when we are expected to override it?

Reply:
I actually had to investigate this and I did see examples in the abstract pep doing this. 

I'm still unconvinced whether it is the right thing to do, tbh, but it is probably OK.

I'm curious what Kapil thinks :)

Reply:
Single underscore is actually equivalent for 'protected', double underscore is 'private'. So I feel like it is ok to override